HOC.md

```diff
@@ -137,7 +137,7 @@ interface WithDataProps<T> {
 // C is the actual interface of the wrapped component (used to grab defaultProps from it)
 export function withSubscription<T, P extends WithDataProps<T>, C>(
   // this type allows us to infer P, but grab the type of WrappedComponent separately without it interfering with the inference of P
-  WrappedComponent: JSXElementConstructor<P> & C,
+  WrappedComponent: React.JSXElementConstructor<P> & C,
```

Comment:
Just a matter of consistency. Sometimes we use property access of the React namespaces, sometimes we assume a named import. Not sure if it's actually ok to use named imports for non-esmodules in the spec or if this only works when using bundlers.